packages/vscode-extension/src/connectionService.ts

```diff
@@ -44,52 +51,79 @@ export function getConnection(key: string): Connection | null {
 export async function deleteConnection(key: string): Promise<void> {
   const connections = getConnections();
   const connection = connections[key];
-  if (connection) {
-    delete connections[key];
-    await updateConnections(connections);
-    await deletePassword(key);
-    await sendNotification('connectionDeleted', {
-      ...connection,
-      connect: false,
-    });
+  if (!connection) {
+    return;
   }
+
+  delete connections[key];
+  await updateConnections(connections);
+  await deletePassword(key);
+  await updateDbmsConnection('connectionDeleted', {
+    ...connection,
+    connect: false,
+  });
 }
 
-export async function saveConnection(
+export function saveConnection(
+  connection: Connection | null,
+): Promise<ConnnectionResult>;
+export function saveConnection(
   connection: Connection | null,
   password: string,
-): Promise<void> {
+): Promise<ConnnectionResult>;
```

Comment:
Probably this is a stupid question. Why do these two functions not have a body?

Reply:
I wanted to overload the saveConnection function

but looking at it.. I can just make the password parameters optional..

Reply:
this is resolved, I've removed the empty overloads